### ISSUE TITLE
Auth issues

### ISSUE THREAD

**Issue by CarlosV89** [aws/aws-amplify#1009]
Do you want to request a feature or report a bug?

What is the current behavior?
Im getting a JWT with AWS Amplify,  trying to verify said JWT but the `kid` in the header is not in my public list; `'https://cognito-idp.' + region + '.amazonaws.com/' + userpool_id + '/.well-known/jwks.json';`

What is the expected behavior?
`kid` should be among the jwks?

**Comment by powerful23** [comment]
@CarlosV89 can you provide more info on this issue like how did you verify and what's the `kid` in the headers?

**Comment by CarlosV89** [comment]
I've fixed it but I'll document what was happening; When I authenticated using the react app, I used the network tab to grab the tokens, the browser was not displaying the whole jwt strings and the result, was an actual jwt that was malformed, I grabbed the response JWT and decoded it to get the idToken/accessToken/refreshToken trio, which then I tried validating and it worked since it now had the correct `kid` in the header.

It was totally my fault. Thanks for the fast response guys! Sry!
